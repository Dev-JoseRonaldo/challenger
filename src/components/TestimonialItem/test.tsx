import { render, screen } from '@testing-library/react'

const { getByText } = screen

import { TestimonialItem, TestimonialItemProps } from '.'

const data: TestimonialItemProps = {
  id: '0',
  author: 'BETH RAYLAND',
  message:
    'Lorem ipsum dolor sit amet consectetur. Enim ac id sapien volutpat.',
}

describe('<TestimonialItem />', () => {
  it('should render the TestimonialItem', () => {
    render(<TestimonialItem {...data} />)

    const author = getByText('BETH RAYLAND')
    const message = getByText(
      'Lorem ipsum dolor sit amet consectetur. Enim ac id sapien volutpat.',
    )

    expect(author).toBeInTheDocument()
    expect(message).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<TestimonialItem {...data} />)
    expect(container).toMatchSnapshot()
  })
})
