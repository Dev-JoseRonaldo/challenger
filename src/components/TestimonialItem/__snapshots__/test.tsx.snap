// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestimonialItem /> should match snapshot 1`] = `
<div>
  <blockquote
    class="text-sm sm:text-xl lg:text-lg xl:text-xl font-medium text-gray-500"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur. Enim ac id sapien volutpat.
    </p>
    <footer
      class="text-primary uppercase font-semibold text-base sm:text-xl tracking-wider mt-5"
    >
      - 
      <cite
        class="not-italic"
      >
        BETH RAYLAND
      </cite>
    </footer>
  </blockquote>
</div>
`;
